aws/data_source_aws_glue_data_catalog_encryption_settings.go

```diff
@@ -0,0 +1,57 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceAwsGlueDataCatalogEncryptionSettings() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceAwsGlueDataCatalogEncryptionSettingsRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"connection_password_encrypted": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"connection_password_kms_key_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"encryption_mode": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"encryption_kms_key_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsGlueDataCatalogEncryptionSettingsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).glueconn
+	input := &glue.GetDataCatalogEncryptionSettingsInput{
+		CatalogId: aws.String(d.Id()),
+	}
+	out, err := conn.GetDataCatalogEncryptionSettings(input)
+	if err != nil {
+		return diag.Errorf("Error reading Glue Data Catalog Encryption Settings: %s", err)
```

Comment:
Capture error context:

```suggestion
		return diag.Errorf("Error reading Glue Data Catalog Encryption Settings: %w", err)
```

Reply:
Making this change and all of the other %w changes causes this error during the unit test:

````
# github.com/terraform-providers/terraform-provider-aws/aws
.\data_source_aws_glue_connection.go:46:10: Errorf call has error-wrapping directive %w
.\data_source_aws_glue_connection.go:57:10: Errorf call has error-wrapping directive %w
.\data_source_aws_glue_data_catalog_encryption_settings.go:48:10: Errorf call has error-wrapping directive %w

Compilation finished with exit code 2
````

Reply:
@danu165, my mistake. Didn't realize that `diag.Errorf` doesn't accept _%w_.